### PR TITLE
Feat: Fixed Search Bar feature with search button successfully Issue 362

### LINKED ISSUE #362: [Bug]: Reloading of posts due to searching
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Even user write the single word the posts are reloading it should dynamically search it instead of showing the preloader after each alphabet typed.

### Expected behavior

Posts reloaded by preloader.

### Add ScreenShots

image

### On which device are you experiencing this bug?

_No response_

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by MeetDOD**
# Pull Request Resolve [#362 ]

### Description

1.  So, instead of calling API again and again i created a button for search so that the api hit doesn't happen when writing each alphabet by user.
2.  It will search only when the user click on the search button.
3.  It will reduce the load on the backend.
4.  The Search/button is responsive and work in light/dark mode.

### Related Issues
Fixes #362 

### Changes Made
New features: Fixed Search Bar feature with search button.

### Screenshots

image

image

image

I certify that I have carried out the relevant code of conduct and provided the requisite screenshot for validation by submitting this pull request.

Thank You for this contribution.

### CODE REVIEW

**Comment by MeetDOD** [comment]
> overall looking, just don't push changes of your other pr

Sorry It want happen again 🙏

File: frontend/src/App.tsx
[the remark this answers is not in the document]
Comment (MeetDOD) [reply frontend/src/App.tsx:1]: Which ?

**Comment by MeetDOD** [comment]
@akbatra567 @Ultimateutkarsh11 Please review this pr, if any issue let me know 😄

**Review by akbatra567 — CHANGES_REQUESTED** [review]
I see there are some other changed in the PR, regarding adding captcha.

Could you remove them from this PR?

**Comment by MeetDOD** [comment]
> I see there are some other changed in the PR, regarding adding captcha.
> 
> Could you remove them from this PR?

You merge the captcha PR and i will resolve the merge conflicts here

**Comment by akbatra567** [comment]
> > I see there are some other changed in the PR, regarding adding captcha.
> > 
> > Could you remove them from this PR?
> 
> You merge the captcha PR and i will resolve the merge conflicts here

PRs should be independent of one another, you can create multiple branches to divide code for each issue. If the other PR is not approved and this gets approved. It's not a good practice to have unrelated code and unapproved changes from the other PR might go unreviewed.

**Comment by MeetDOD** [comment]
@akbatra567 approve/review my other PR i will fix this one now only !

**Comment by MeetDOD** [comment]
Done

**Comment by MeetDOD** [comment]
@akbatra567 please review this PR 🙏 @Ultimateutkarsh11

File: frontend/src/pages/Signin.tsx [hunk 1]
```diff
@@ -33,6 +35,11 @@ const Signin = () => {
       return;
     }
 
+    if (!isCaptchaValid) {
+      toast.error('Captcha is not valid');
+      return;
+    }
+
```
Comment (akbatra567) [inline frontend/src/pages/Signin.tsx:42]: Remove this duplicate code

File: frontend/src/pages/Posts.tsx [hunk 2]
```diff
@@ -77,6 +79,10 @@ const Posts = () => {
     }
   };
 
+  const handleSearch = () => {
+    fetchPosts(page, 12, searchQuery, filterTags);
```
Comment (akbatra567) [inline frontend/src/pages/Posts.tsx:83]: make page size configurable.

Potential solution:
1. Have a drop down on UI to select page size. (if we already have that in UI)
2. Store it environment variable and fetch from there.

Comment (MeetDOD) [reply frontend/src/pages/Posts.tsx:83]: I haven't mention this in the PR did I ?

File: frontend/src/pages/Signin.tsx (hunk 1 above)
Comment (MeetDOD) [reply frontend/src/pages/Signin.tsx:42]: Ok 👍

Comment (MeetDOD) [reply frontend/src/pages/Signin.tsx]: Done @akbatra567

File: frontend/src/pages/Signin.tsx [hunk 3]
```diff
@@ -108,12 +110,13 @@ const Signin = () => {
...
               )}
               </span>
-            </label>               
+            </label>      
             <p className="text-sm font-semibold mb-2 text-red-600">
               {error.password}
             </p>
             <CaptchaUser onChange={(isValid) => setIsCaptchaValid(isValid)} /> {/* Add Captcha component */}
           </div>
+          <CaptchaUser onChange={(isValid) => setIsCaptchaValid(isValid)} />
```
Comment (VaibhavArora314) [inline frontend/src/pages/Signin.tsx:119]: remove captcha from here since already present

**Review by VaibhavArora314 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Signin.tsx (hunk 3 above)
Comment (MeetDOD) [reply frontend/src/pages/Signin.tsx:117]: Done @VaibhavArora314

**Review by VaibhavArora314 — APPROVED** [review]

**Comment by MeetDOD** [comment]
@Ultimateutkarsh11 Please review this pr if changes exists let me know :)

**Review by Ultimateutkarsh11 — APPROVED** [review]

**Review by Ultimateutkarsh11 — APPROVED** [review]

**Comment by MeetDOD** [comment]
@akbatra567 Thanks and also check this also 👍

**Comment by MeetDOD** [comment]
@akbatra567 Thanks and also check this also 👍

File: frontend/src/pages/Posts.tsx (hunk 2 above)
Comment (akbatra567) [reply frontend/src/pages/Posts.tsx:83]: Let's ignore this for now then.

**Review by akbatra567 — APPROVED** [review]

**Comment by MeetDOD** [comment]
@VaibhavArora314 merge this pr and have a look at pr #385

**Pull request merged**